README.md

```diff
@@ -211,19 +211,54 @@ All pushes are done via HTTP. The interface is vaguely REST-like.
 
 The default port the push gateway is listening to is 9091. The path looks like
 
-    /metrics/job/<JOBNAME>{/<LABEL_NAME>/<LABEL_VALUE>}
+    /metrics/job/<JOB_NAME>{/<LABEL_NAME>/<LABEL_VALUE>}
 
-`<JOBNAME>` is used as the value of the `job` label, followed by any
+`<JOB_NAME>` is used as the value of the `job` label, followed by any
 number of other label pairs (which might or might not include an
 `instance` label). The label set defined by the URL path is used as a
 grouping key. Any of those labels already set in the body of the
 request (as regular labels, e.g. `name{job="foo"} 42`)
 _will be overwritten to match the labels defined by the URL path!_
 
-Note that `/` cannot be used as part of a label value or the job name,
-even if escaped as `%2F`. (The decoding happens before the path
-routing kicks in, cf. the Go documentation of
-[`URL.Path`](http://golang.org/pkg/net/url/#URL).)
+If `job` or any label name is suffixed with `$base64`, the following job name
+or label value is interpreted as a base64 encoded string according to [RFC
+4648, using the URL and filename safe
+alphabet](https://tools.ietf.org/html/rfc4648#section-5). (Padding is
+optional.) This is the only way of using job names or label values that contain
+a `/`. For other special characters, the usual URI component encoding works,
+too, but the base64 might be more convenient.
+
+Ideally, client libraries take care of the suffixing and encoding.
+
+Examples:
+
+* To use the grouping key `job="directory_cleaner",path="/var/tmp"`, the
+  following path will _not_ work:
+
+	  /metrics/job/directory_cleaner/path//var/tmp
+	  
+  Instead, use the base64 URL-safe encoding for the label value and mark it by
+  suffixing the label name with `$base64`:
+  
+  	  /metrics/job/directory_cleaner/path$base64/L3Zhci90bXA
+	  
+  If you are not using a client library that handles the encoding for you, you
+  can use encoding tools. For example, there is a command line tool `base64url`
+  (Debian package `basez`), which you could combine with `curl` to push from
+  the command line in the following way:
+  
+      echo 'some_metric{foo="bar"} 3.14' | curl --data-binary @- http://pushgateway.example.org:9091/metrics/job/directory_cleaner/path\$base64/$(echo -n '/var/tmp' | base64url)
+    
+  Remember to escape the `$` in `$base64`.
```

Comment:
Hmm, I wonder if we should use a different character to avoid this problem.

Reply:
I mostly picket it because JetBrains apparently use `$base64:` somewhere in their restful APIs as a marker. But that's far from being a de-facto standard. Our marker character should just be one that doesn't need fancy URL-encoding. `@` was my second candidate on the list. Would you like that more?

Reply:
Yeah, @ would cause fewer problems.

Reply:
OK, changing it and will merge after.